kotlintest-assertions/src/commonMain/kotlin/io/kotlintest/properties/gens.kt

```diff
@@ -585,3 +585,26 @@ fun Gen.Companion.factors(k: Int): Gen<Int> = object : Gen<Int> {
          .filter { k % it == 0 }
    }
 }
+
+/**
+ * Returns a [[Gen]] which returns the sample values in same order as they are passed in, once all sample values are used
+ * it repeats elements from starting.
+ */
+fun <T:Any> Gen.Companion.samples(vararg sampleValues: T) = object : Gen<T> {
+    private fun nextNumberGenerator(): () -> T  {
+        var currentIndex = 0;
+        return {
```

Comment:
Any reason this return statement opens a `{` instead of returning `nextValue` at the end?

Reply:
That's because generateInfiniteSequence needs a lambda to generate the next element. That's the reason I am returning a lambda instead of the next element. I am changing its name to reflect that.

Reply:
Oh, right